packages/auth/src/core/auth/register.ts

```diff
@@ -63,36 +63,38 @@ export function registerAuth(clientPlatform: ClientPlatform): void {
         const app = container.getProvider('app').getImmediate()!;
         const heartbeatServiceProvider =
           container.getProvider<'heartbeat'>('heartbeat');
+        const appCheckServiceProvider =
+          container.getProvider<'app-check-internal'>('app-check-internal');
         const { apiKey, authDomain } = app.options;
-        return ((app, heartbeatServiceProvider) => {
-          _assert(
-            apiKey && !apiKey.includes(':'),
-            AuthErrorCode.INVALID_API_KEY,
-            { appName: app.name }
-          );
-          // Auth domain is optional if IdP sign in isn't being used
-          _assert(!authDomain?.includes(':'), AuthErrorCode.ARGUMENT_ERROR, {
-            appName: app.name
-          });
-          const config: ConfigInternal = {
-            apiKey,
-            authDomain,
-            clientPlatform,
-            apiHost: DefaultConfig.API_HOST,
-            tokenApiHost: DefaultConfig.TOKEN_API_HOST,
-            apiScheme: DefaultConfig.API_SCHEME,
-            sdkClientVersion: _getClientVersion(clientPlatform)
-          };
 
-          const authInstance = new AuthImpl(
-            app,
-            heartbeatServiceProvider,
-            config
-          );
-          _initializeAuthInstance(authInstance, deps);
+        _assert(
```

Comment:
Why are we getting rid of the anon function here? I think we copied this from Firestore originally (not exactly sure why it's done this way)

Reply:
Yeahh, I wasn't sure why we had the anon function either and didn't notice a difference in behavior with / without it? I think Firestore currently doesn't use an anon function: https://github.com/firebase/firebase-js-sdk/blob/825e648b81ca63c7bc64f8700f7a46eb320b2106/packages/firestore/src/register.ts#L45

Will leave this open if other folks have an opinion about it

Reply:
+1 to remove the anon function if we did not observe any behavior change